### PR TITLE
Adding shared pool monitoring for Oracle databases

### PR DESCRIPTION
**Pull request by nenadnoveljic**
### What does this PR do?

Add shared memory monitoring (a.k.a system global area - SGA) for Oracle databases.

### Motivation

Shared memory consists of different pools that store different types of data: SQL cache, buffer cache, JVM cache, etc. The cache sizes are dynamic by default. The growth and shrinking provides valuable information for performance analysis.

### Describe how to test/QA your changes

Check if shared memory is correctly displayed in frontend.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by cswatt — APPROVED** [review]
good for docs

File: pkg/collector/corechecks/oracle-dbm/shared_memory.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	if err != nil {
+		return fmt.Errorf("failed to collect shared memory info: %w", err)
+	}
+	sender, err := c.GetSender()
+	if err != nil {
+		return fmt.Errorf("failed to initialize sender: %w", err)
+	}
+	for _, r := range rows {
+		tags := appendPDBTag(c.tags, r.PdbName)
+		//tags = append(tags, `memory:"`+r.Memory+`"`)
```
Comment (jmeunier28) [inline pkg/collector/corechecks/oracle-dbm/shared_memory.go:42]: can we remove this line?

File: releasenotes/notes/oracle-shared-memoy-06e54ebb7173317b.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    Add shared memory (a.k.a. system global area - SGA) for Oracle databases.
```
Comment (jmeunier28) [inline releasenotes/notes/oracle-shared-memoy-06e54ebb7173317b.yaml:11]: if we are adding a new metric, should we also document that in the release notes?

File: pkg/collector/corechecks/oracle-dbm/shared_memory.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		memoryTag := strings.ReplaceAll(r.Memory, " ", "_")
+		memoryTag = strings.ToLower(memoryTag)
+		memoryTag = strings.ReplaceAll(memoryTag, "_size", "")
+		tags = append(tags, fmt.Sprintf("memory:%s", memoryTag))
```
Comment (jmeunier28) [inline pkg/collector/corechecks/oracle-dbm/shared_memory.go:46]: what are the possible values for this?

**Review by jmeunier28 — APPROVED** [review]
looks fine to me, just left a few questions

File: releasenotes/notes/oracle-shared-memoy-06e54ebb7173317b.yaml (hunk 2 above)
Comment (alexandre-normand) [reply releasenotes/notes/oracle-shared-memoy-06e54ebb7173317b.yaml:11]: The documentation aspect is a good question. We talked a bit about this on a previous PR and there doesn't appear to be anything like the metadata.csv that we have in agent integrations for the core agent checks. I wonder if we should still add a metadata.csv here even if it ends up not being parsed to generate documentation pages.

**Review by alexandre-normand — APPROVED** [review]

File: pkg/collector/corechecks/oracle-dbm/shared_memory.go (hunk 1 above)
Comment (nenadnoveljic) [reply pkg/collector/corechecks/oracle-dbm/shared_memory.go:42]: 👍

File: releasenotes/notes/oracle-shared-memoy-06e54ebb7173317b.yaml (hunk 2 above)
Comment (nenadnoveljic) [reply releasenotes/notes/oracle-shared-memoy-06e54ebb7173317b.yaml:11]: I added the metric to the release notes. I'll create  `metadata.csv` at some point. We're generally lagging with documentation and will do a better job in the future. For the moment, I'm focusing to fill the feature gaps that are relevant to prospects.

Comment (nenadnoveljic) [reply releasenotes/notes/oracle-shared-memoy-06e54ebb7173317b.yaml:11]: 👍

File: pkg/collector/corechecks/oracle-dbm/shared_memory.go (hunk 3 above)
Comment (nenadnoveljic) [reply pkg/collector/corechecks/oracle-dbm/shared_memory.go:46]:
```
  1  SELECT
  2	 s.name
  3    FROM v$sgainfo s
  4    WHERE
  5*	  s.name NOT IN ('Maximum SGA Size','Startup overhead in Shared Pool','Granule Size','Shared IO Pool Size')
SQL> /

NAME
--------------------------------
Fixed SGA Size
Redo Buffers
Buffer Cache Size
In-Memory Area Size
Shared Pool Size
Large Pool Size
Java Pool Size
Streams Pool Size
Data Transfer Cache Size
Free SGA Memory Available
```

**Pull request merged**